### PR TITLE
(#20) Cleanup matcher

### PR DESCRIPTION
**Pull request by victornoel**
This is for #20 that was incorrectly closed.

### CODE REVIEW

File: src/test/java/org/llorllale/cactoos/matchers/MatchesTest.java [hunk 1]
```diff
@@ -58,41 +54,15 @@ void matches() {
      * Gives negative testing result for the invalid arguments.
      */
     @Test
-    void matchStatus() {
+    void mismatches() {
```
Comment (baudoliver7) [inline src/test/java/org/llorllale/cactoos/matchers/MatchesTest.java:57]: @victornoel What about the match case.

**Review by baudoliver7 — CHANGES_REQUESTED** [review]
@victornoel Please, see comments below.

File: src/main/java/org/llorllale/cactoos/matchers/RunsInThreads.java [hunk 2]
```diff
@@ -46,7 +47,7 @@
  * @since 0.24
  * @checkstyle JavadocMethodCheck (500 lines)
  */
-public final class RunsInThreads<T> extends TypeSafeMatcher<Func<T, Boolean>> {
+public final class RunsInThreads<T> extends TypeSafeDiagnosingMatcher<Func<? super T, Boolean>> {
```
[the remark this answers is not in the document]
Comment (victornoel) [reply src/main/java/org/llorllale/cactoos/matchers/RunsInThreads.java:50]: @baudoliver7 yep, I suppose it's not really needed here, I will remove it to not make things over complex

File: src/test/java/org/llorllale/cactoos/matchers/MatchesTest.java (hunk 1 above)
Comment (victornoel) [reply src/test/java/org/llorllale/cactoos/matchers/MatchesTest.java:57]: @baudoliver7 I will suppose you mean testing `Matches` using `Matches`? it's a good idea

**Comment by victornoel** [comment]
@baudoliver7 thank you, it's done

File: src/test/java/org/llorllale/cactoos/matchers/MatchesTest.java (hunk 1 above)
Comment (baudoliver7) [reply src/test/java/org/llorllale/cactoos/matchers/MatchesTest.java:66]: @victornoel Yes

**Comment by baudoliver7** [comment]
@victornoel thx, looks good to me :)

**Comment by victornoel** [comment]
@rultor merge

**Comment by victornoel** [comment]
@rultor merge

**Comment by victornoel** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 24 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @baudoliver7[/z](https://www.0crat.com/u/baudoliver7), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Comment by sereshqua** [comment]
@baudoliver7 please make sure you will find at least 3 issues during next CR, thanks

**Pull request merged**